Improve docs around pipeline target types (#3318)
FlowFuse/flowfuse: docs/user/device-groups.md

---
navTitle: Device Groups
---

# Device Groups

**Navigation**: `Team > Application > Device Groups`

> Device groups allow you to organise your Application assigned devices into logical groups.

Reviewer: I believe the _why_ a user wants this isn't well covered.

When managing many devices running the same snapshot, ..... etc etc

> These groups can be the target of [DevOps Pipelines](./devops-pipelines.md), greatly

Reviewer: Is there another way to deploy to these groups? `can` is implying that for me. In other words, is there tight coupling with devops pipelines or not?

simplifying deployments to one or hundreds of devices.

The following requirements apply:

 - FlowFuse 1.15+ running with an EE license

>  - FlowFuse Cloud

Author: Device groups are Enterprise right? - can you suggest suitable wording please @ZJvandeWeg

>  - FlowFuse Cloud

Reviewer 2: Running with an EE license is difference to a given Team Tier. FF Cloud runs on an EE License, but only Enterprise Teams have access to Device Groups iirc?

>  - FlowFuse Cloud

Author: Now updated to:

> The following requirements apply:
> 
>  - FlowFuse 1.15+ Enterprise tier
> - FlowFuse Cloud

Please offer alternative suggestion if not ok

@ZJvandeWeg @joepavitt

>  - FlowFuse Cloud

Reviewer 3: @Steve-Mcl a more correct wording is:

> - FlowFuse 1.15+ running with an EE license
> - FlowFuse Cloud Enterprise Teams

>  - FlowFuse Cloud

Reviewer 2: Will leave it to ZJ to work on this, I _believe_ he's asking for "FlowFuse Cloud - Enterprise Team Tier", but I'm not 100% certain

>  - FlowFuse Cloud

Author: Updated to 

> The following requirements apply:
> - FlowFuse 1.15+ running with an EE license
> - FlowFuse Cloud Enterprise Teams

Thanks @knolleary

## Creating a Device Group

> Before starting, ensure you have 2 or more devices assigned to your application.

Author: You can, if you wish, create a group, then go create devices, then come back and update them, but for the sake of simplicity I added that line - however i completely understand the point you are making. I will revisit / revise.

> Before starting, ensure you have 2 or more devices assigned to your application.

Reviewer 2: Yeah, to echo @ZJvandeWeg here - the "ensure you have 2 or more devices" is odd phrasing, surely the minimum is 1, if any at all?

> You can find out how to do that [here](../device-agent/register.md#assign-the-device-to-an-application)

Author: removed whole sentence (not relevant)

1. Select the Application you want to configure a Device Group for.
1. Select the "Device Groups" tab
1. Click "Add Device Group"
1. Name your Device Group appropriately (this can be changed later)
1. Click "Create"

_Note: Adding a description can help you better distinguish device groups._

## Updating Device Group Membership

1. In the Device Groups table, click the Device Group you want to modify
1. Click "Edit"
   1. On the left, you will be shown available devices (ones that are assigned to your application and are available to be added to a device group)
   1. On the right, you will be shown devices that are already in the device group
1. Place a checkmark next to the devices in the Available Devices list that you want to add to the Device Group then click "Add Devices"
1. Place a checkmark next to the devices in the Device Group list that you want to remove then click "Remove Devices"

> 1. Click "Save" to commit your changes

Reviewer: What happens after? What if I added a device to a group and the group is supposed to run snapshot foobar, does it run foobar too?

Author: That will be added in the PR on branch [3260-update-device-on-group-membership-change](https://github.com/FlowFuse/flowfuse/tree/3260-update-device-on-group-membership-change) and why I am holding off raising that PR until this PR is merged.

_Note: If you make a mistake, you can cancel your changes at any time by clicking "Cancel"_

> _Note: You can only add devices to a device group that are assigned to the application and are not already in another device group._

Reviewer: ```suggestion
_Note: When a device you want to add to a group doesn't appear in the list, it's likely already assigned to another group._
```

---
navTitle: Device Groups
---

# Device Groups

**Navigation**: `Team > Application > Device Groups`

<img src="./images/ui-device-groups.png" width="100%" />

When managing many devices that are intended to run the same [snapshot](./snapshots.md), Device Groups allow you
to organise your devices into logical groups.
These groups can then be set as the target of a [DevOps Pipelines](./devops-pipelines.md).
This greatly simplifying deployments of the same configuration to one or even hundreds of devices with a single click.

The following requirements apply:

> - FlowFuse 1.15+ running with an EE license

Reviewer: @Steve-Mcl There's no EE license, it doesn't exist. It's the Enterprise tier.

Author: @ZJvandeWeg this is coflicting with @knolleary suggestion: https://github.com/FlowFuse/flowfuse/pull/3318#discussion_r1449053434

Please advise.

Reviewer 3: @Steve-Mcl it isn't conflicting - its an improved iteration. 

```suggestion
- FlowFuse 1.15+ Enterprise Tier
```

> - FlowFuse Cloud Enterprise Teams

Reviewer: ```suggestion
- FlowFuse Cloud teams on the Enterprise tier
```
